cirq-superstaq/cirq_superstaq/job.py

```diff
@@ -201,9 +306,27 @@ def counts(self, timeout_seconds: int = 7200, polling_seconds: float = 1.0) -> D
             time_waited_seconds += polling_seconds
```

Comment:
Will look into it in another PR since this one is close to 500 lines for review (seems a bit related to https://github.com/Infleqtion/client-superstaq/issues/545 also)